test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Sia.Gateway.Controllers;
+using Sia.Shared.Protocol;
+using System.Collections.Generic;
+
+namespace Sia.Gateway.Tests.Requests.Events
+{
+    [TestClass]
+    public class GenerateLinksHeaderTestsForEvents
+    {
+        private List<string> methods;
+        private List<object> ids;
+        private Mock<IUrlHelper> urlHelperMock;
+        public GenerateLinksHeaderTestsForEvents()
+        {
+            // Arrange
+            methods = new List<string>();
+            ids = new List<object>();
+            urlHelperMock = new Mock<IUrlHelper>();
+            urlHelperMock.Setup(link => link.Link(It.IsAny<string>(), It.IsAny<object>()))
+                .Callback<string, object>(
+                    (s, o) =>
+                    {
+                        methods.Add(s);
+                        ids.Add(o);
+                    }
+                );
+
+        }
+
+        [TestMethod]
+        public void CreateLinksWorksCorrectly_WhenCalledInEventsMethods()
+        {
+            //Arrange
+            methods.Clear();
+            ids.Clear();
+            var eventsController = new EventsController(null, null, null, urlHelperMock.Object);
+            // Act
+            eventsController.CreateLinks("1", "2", null,null,"");
+
+            // Assert
+            urlHelperMock.Verify(foo => foo.Link(EventsController.GetSingleRouteName, It.IsAny<object>()), Times.Exactly(1));
+            urlHelperMock.Verify(foo => foo.Link(EventsController.PostSingleRouteName, It.IsAny<object>()), Times.Exactly(1));
+            urlHelperMock.Verify(foo => foo.Link(EventsController.GetMultipleRouteName, It.IsAny<object>()), Times.Exactly(1));
+            urlHelperMock.Verify(foo => foo.Link(IncidentsController.GetSingleRouteName, It.IsAny<object>()), Times.Exactly(1));
+
+            Assert.AreEqual(ids[0].ToString(), "{ id = 1 }");
+            //Assert.AreEqual(ids[0], new { id = "1" });
+            Assert.AreEqual(ids[1].ToString(), "{ }");
+            Assert.AreEqual(ids[2].ToString(), "{ }");
+            Assert.AreEqual(ids[3].ToString(), "{ id = 2 }");
```

Comment:
Looking closer at both CreateLinks methods I noticed that their argument signature is similar, but not the same, and they are. Design-wise I would prefer that the accepted arguments were the same, along with the format of the return values, though I think this sort of nitpickiness is overkill.